repo/config/version.go

```diff
@@ -1,9 +0,0 @@
-package config
-
-// CurrentCommit is the current git commit, this is set as a ldflag in the Makefile
-var CurrentCommit string
-
-// CurrentVersionNumber is the current application's version literal
-const CurrentVersionNumber = "0.4.17-dev"
```

Comment:
Let's find a way to not have the binary version in a different package. I really don't think this stuff needs to be in the config package anyways

Reply:
Yep, that definitely belongs here.

The problem is that we apparently embed that in config on init:
https://github.com/ipfs/go-ipfs/blob/80fcc4ecf177b3b6a8daa10559eb46bcf8a931db/repo/config/init.go#L27-L32

This was introduced in https://github.com/ipfs/go-ipfs/commit/feef5c3415ec249cf307f43a422545de24aee319#diff-22f880839db0df9b05dcad0718392fc9R28 which is in 0.4.16.

From what I can see we can safely drop that as in the only place it's used it's overridden with https://github.com/ipfs/go-ipfs/blob/master/core/corehttp/commands.go#L80

Reply:
That sounds good to me.